extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailClientProducer.java

```diff
@@ -16,52 +20,74 @@
-            LOGGER.warn(
-                    "`io.vertx.axle.ext.mail.MailClient` is deprecated and will be removed in a future version - it is "
-                            + "recommended to switch to `io.vertx.mutiny.ext.mail.MailClient`");
-            axleMailClient = io.vertx.axle.ext.mail.MailClient.newInstance(client);
-        }
-        return axleMailClient;
+    public io.vertx.axle.ext.mail.MailClient axleMailClient() {
+        LOGGER.warn(
+                "`io.vertx.axle.ext.mail.MailClient` is deprecated and will be removed in a future version - it is "
+                        + "recommended to switch to `io.vertx.mutiny.ext.mail.MailClient`");
+        return io.vertx.axle.ext.mail.MailClient.newInstance(client);
     }
 
     @Singleton
     @Produces
     @Deprecated
-    public synchronized io.vertx.reactivex.ext.mail.MailClient rxMailClient() {
-        if (rxMailClient == null) {
-            LOGGER.warn(
-                    "`io.vertx.reactivex.ext.mail.MailClient` is deprecated and will be removed in a future version - it is "
-                            + "recommended to switch to `io.vertx.mutiny.ext.mail.MailClient`");
-            rxMailClient = io.vertx.reactivex.ext.mail.MailClient.newInstance(client);
-        }
-        return rxMailClient;
+    public io.vertx.reactivex.ext.mail.MailClient rxMailClient() {
+        LOGGER.warn(
+                "`io.vertx.reactivex.ext.mail.MailClient` is deprecated and will be removed in a future version - it is "
+                        + "recommended to switch to `io.vertx.mutiny.ext.mail.MailClient`");
+        return io.vertx.reactivex.ext.mail.MailClient.newInstance(client);
+    }
+
+    @PreDestroy
+    public void stop() {
+        client.close();
+    }
+
+    private MailClient mailClient(Vertx vertx, MailConfig config) {
+        io.vertx.ext.mail.MailConfig cfg = toVertxMailConfig(config);
+        return MailClient.createShared(vertx, cfg);
+    }
+
+    private io.vertx.ext.mail.MailConfig toVertxMailConfig(MailConfig config) {
+        io.vertx.ext.mail.MailConfig cfg = new io.vertx.ext.mail.MailConfig();
+        config.authMethods.ifPresent(cfg::setAuthMethods);
+        cfg.setDisableEsmtp(config.disableEsmtp);
+        cfg.setHostname(config.host);
+        cfg.setKeepAlive(config.keepAlive);
+        config.keyStore.ifPresent(cfg::setKeyStore);
+        config.keyStorePassword.ifPresent(cfg::setKeyStorePassword);
+        config.login.ifPresent(s -> cfg.setLogin(LoginOption.valueOf(s.toUpperCase())));
+        config.maxPoolSize.ifPresent(cfg::setMaxPoolSize);
+        config.ownHostName.ifPresent(cfg::setOwnHostname);
+        config.username.ifPresent(cfg::setUsername);
+        config.password.ifPresent(cfg::setPassword);
+        config.port.ifPresent(cfg::setPort);
+        cfg.setSsl(config.ssl);
+        config.startTLS.ifPresent(s -> cfg.setStarttls(StartTLSOptions.valueOf(s.toUpperCase())));
+        cfg.setTrustAll(config.trustAll);
```

Comment:
Let's drop the lambdas while we're at it?

Reply:
Sure, I'll do that once I get back to this PR :)